### PR TITLE
GODRIVER-1361 enable MgoRegistry

### PR DESCRIPTION
**Pull request by iwysiu**

### CODE REVIEW

File: bson/mgocompat/registry.go [hunk 1]
```diff
@@ -31,17 +31,17 @@ var (
 	tSetter         = reflect.TypeOf((*Setter)(nil)).Elem()
 )
 
-// mgoRegistry is the mgo compatible bsoncodec.Registry. It contains the default and
+// MgoRegistry is the mgo compatible bsoncodec.Registry. It contains the default and
 // primitive codecs with mgo compatible options.
-var mgoRegistry = newMgoRegistryBuilder().Build()
+var MgoRegistry = NewMgoRegistryBuilder().Build()
```
Comment (craiggwilson) [inline bson/mgocompat/registry.go]: I would call this just Registry. We are already in the `mgocompat` package, so I think `mgocompat.Registry` is clear enough.

Same with the naming below. No need to repeat `mgo` everything I don't think.

File: bson/mgocompat/registry.go [hunk 2]
```diff
@@ -86,9 +86,10 @@ func newMgoRegistryBuilder() *bsoncodec.RegistryBuilder {
 	return rb
 }
 
-// buildMgoRNVRegistry creates a new bsoncodec.RegistryBuilder configured to behave like mgo/bson with RespectNilValues set to true.
-func newMgoRespectNilValuesRegistryBuilder() *bsoncodec.RegistryBuilder {
-	rb := newMgoRegistryBuilder()
+// NewMgoRespectNilValuesRegistryBuilder creates a new bsoncodec.RegistryBuilder configured to behave like mgo/bson
+// with RespectNilValues set to true.
+func NewMgoRespectNilValuesRegistryBuilder() *bsoncodec.RegistryBuilder {
```
Comment (craiggwilson) [inline bson/mgocompat/registry.go]: What is the default behaviour of `mgo` with respect to this?

Comment (iwysiu) [reply bson/mgocompat/registry.go]: mgo's default behavior is RespectNilValues=false

File: bson/mgocompat/registry.go (hunk 1 above)
Comment (iwysiu) [reply bson/mgocompat/registry.go]: done

**Pull request merged**